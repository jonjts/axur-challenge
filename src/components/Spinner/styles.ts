import styled from 'styled-components';

export const Container = styled.div<{ size: string, color?: string }>`
    border: 4px solid #f3f3f3; 
    border-top: 4px solid ${({ color, theme: { colors: { secondary } } }) => color || secondary};
    border-radius: 50%;
    width: ${({ size }) => size};
    height: ${({ size }) => size};
    animation: spinner 2s linear infinite;
`;
