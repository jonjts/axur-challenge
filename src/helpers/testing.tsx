import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import ThemeProvider from '../providers/Theme';
import InspectionProvider from '../providers/Inspection'

export const putProviders = (component: JSX.Element) => (
    <ThemeProvider>
        <InspectionProvider>
            {component}
        </InspectionProvider>
    </ThemeProvider>
);

export const matchSnapshotWithProviders = (component: JSX.Element) => {
    const tree = renderer.create(putProviders(component)).toJSON();
    expect(tree).toMatchSnapshot();
    return tree;
};

export const renderWithProviders = (component: JSX.Element) => {
    return render(putProviders(component));
};
