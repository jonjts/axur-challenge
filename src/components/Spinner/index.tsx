import React, { memo } from 'react';

import { Container } from './styles';

export enum SIZE {
    sm = '20px', md = '50px', lg = '80px'
};

type Props = {
    color?: string,
    size?: SIZE
};

const Spinner: React.FC<Props> = ({ color, size = SIZE.md }) => (
    <Container color={color} size={size} />
);

export default memo(Spinner);