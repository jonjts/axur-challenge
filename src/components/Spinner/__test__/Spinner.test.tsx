import { matchSnapshotWithProviders, renderWithProviders } from '../../../helpers/testing';

import Spinner from '../index';

describe('NavBar', () => {
  test('match with snapshot', () => {
    matchSnapshotWithProviders(<Spinner />);
  });

  test('renders correctly', () => {
    renderWithProviders(<Spinner />);
  });
});
