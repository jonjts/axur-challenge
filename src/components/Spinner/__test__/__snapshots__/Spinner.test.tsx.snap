// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar match with snapshot 1`] = `
<div
  className="sc-gsTCUz oIHdM"
  size="50px"
/>
`;
